### ISSUE TITLE
Error Updating google_compute_service_attachment Resource with connection_preference Set to `ACCEPT_AUTOMATIC`

### ISSUE THREAD

**Issue by mitchcail** [hashicorp/terraform-provider-google#17153]
### Terraform Version
v1.3.7

### Affected Resource(s)
google_compute_service_attachment

### Debug Output

Gist to debug output: https://gist.github.com/mitchcail/5737429f0bf7cb6502f89696136f6439

### Expected Behavior

The google_compute_service_attachment resource should be updated in-place, with connection_preference changing to "ACCEPT_AUTOMATIC" without any errors.

### Actual Behavior

The terraform apply command results in an error:
```
Error: Error updating ServiceAttachment "projects/sandbox/regions/europe/serviceAttachments/sandbox-psc": googleapi: Error 400: Invalid value for field 'resource.connectionPreference': 'ACCEPT_AUTOMATIC'. Incorrect combination of connection preference with accept and/or reject list, invalid
```

### Steps to Reproduce

1. Create a compute_service_attachment and set [connection_preference](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_service_attachment#connection_preference) to `ACCEPT_MANUAL`
2. Update the connection_preference to `ACCEPT_AUTOMATIC`

### More Context
It looks like the PATCH request sends in the `consumerRejectLists` and `consumerAcceptLists` when attempting to update from `ACCEPT_MANUAL` to `ACCEPT_AUTOMATIC`. These should not be included for this PATCH request.

**Comment by edwardmedia** [comment]
Yes, we should not have consumerRejectLists in the payload when it is a empty list.

```
 {
  "connectionPreference": "ACCEPT_AUTOMATIC",
  "consumerAcceptLists": [
   {
    "connectionLimit": 10,
    "projectIdOrNum": "testing-sandbox"
   }
  ],
  "consumerRejectLists": [],
  "description": "sandbox-psc. Configured with terraform",
  "name": "sandbox-psc",
  "natSubnets": [
   "projects/sandbox/regions/europe/subnetworks/sandbox-psc-subnet"
  ]
 }
```

**Comment by TenSt** [comment]
I've encountered the same behavior when trying to create google_compute_service_attachment resource with connection_preference set to "ACCEPT_AUTOMATIC" and different combinations of consumer_accept_lists and consumer_reject_lists. The error is the same: 
```
Error: Error creating ServiceAttachment: googleapi: Error 400: Invalid value for field 'resource.connectionPreference': 'ACCEPT_AUTOMATIC'. Incorrect combination of connection preference with accept and/or reject list, invalid
```

**Comment by huguesalary** [comment]
I'm running into this issue.

It's unclear whether `ACCEPT_AUTOMATIC` is allowed when using an accept list and/or a reject list.

Is the error message above a bug? Or is it a user error?

Can the use of `ACCEPT_AUTOMATIC` be clarified?
